src/components/Avatar/Avatar.tsx

```diff
@@ -21,6 +29,126 @@ export const AVATAR_THEME_SET = [
   styles.grey,
 ];
 
+export const getStatusItemSizeAndPadding = (
+  avatarSize: number
+): [number, number] => {
+  const statusItemSize: number = (avatarSize * 16) / 100;
+  return [statusItemSize, (statusItemSize * 6) / 16];
+};
+
+const statusItemPositionRtlMap: {
+  [key in StatusItemsPosition]: StatusItemsPosition;
+} = {
+  [StatusItemsPosition.Top]: StatusItemsPosition.Top,
+  [StatusItemsPosition.Bottom]: StatusItemsPosition.Bottom,
+  [StatusItemsPosition.Left]: StatusItemsPosition.Right,
+  [StatusItemsPosition.Right]: StatusItemsPosition.Left,
+  [StatusItemsPosition.TopRight]: StatusItemsPosition.TopLeft,
+  [StatusItemsPosition.TopLeft]: StatusItemsPosition.TopRight,
+  [StatusItemsPosition.BottomRight]: StatusItemsPosition.BottomLeft,
+  [StatusItemsPosition.BottomLeft]: StatusItemsPosition.BottomRight,
+};
+
+const AvatarStatusItems: FC<BaseAvatarProps> = React.forwardRef(
+  ({ outline, size, statusItems }) => {
+    const htmlDir: string = useCanvasDirection();
+
+    const getStatusItemPositionStyle = (
+      itemPos: StatusItemsPosition,
+      itemProps: StatusItemsProps
+    ): React.CSSProperties => {
+      const outlineWidth: string = outline ? outline.outlineWidth : '0px'; // Avatar outline width
+      const avatarWidth: string = size;
+      const itemWidth: string = `(${itemProps.size} + (2 * ${itemProps.padding}))`; // Status item width
+
+      switch (htmlDir === 'rtl' ? statusItemPositionRtlMap[itemPos] : itemPos) {
```

Comment:
There should be 2 more cases to support upcoming UX indicator layout requirements:

StatusItemsPosition.BottomStacked
StatusItemsPosition.RightStacked

StatusItemsPosition.BottomStacked (From Figma) -->

<img width="691" alt="bottomStacked" src="https://user-images.githubusercontent.com/99700808/219814839-2815cdd0-ebf3-44a8-9734-7d45fa3e2c2c.png">

StatusItemsPosition.RightStacked (Needs RTL Mapping too) -->

<img width="384" alt="Screenshot 2023-02-17 at 3 32 43 PM" src="https://user-images.githubusercontent.com/99700808/219816551-915a2ff6-1687-4c26-85b7-724f20e4e083.png">

Reply:
@dkilgore-eightfold we need the other changes, this can be added later